### PR TITLE
feat(profile) Add Shop UI + Reward System

### LINKED ISSUE #67: EduMon Reward
###  Task
- [ ] Implement a reward system where completed tasks grant coins.
- [ ] Prepare the logic for future Shop integration.

### PR DESCRIPTION
**Pull request by SamAbd11805**
## Summary

This Pull Request introduces major testing and feature improvements for the EduMon app:  
- Complete ProfileScreen tests (UI + ViewModel logic)  
- Implementation of the ShopScreen UI for cosmetic items  
- Integration of the reward system (coins earned → usable in the shop)  
- Full unit tests for ProfileViewModel and ShopViewModel  
- Global test coverage improved to approximately 85%  

---

### Added
- `ShopScreen.kt` — Fully styled shop with coins, cosmetics, and animations  
- `ShopViewModel.kt` — Handles purchase logic and coin balance updates  
- `ProfileViewModel.kt` — Added reward system (`addCoins()` function)  
- `ProfileScreenTest.kt` — Tests all sections, settings, and callbacks  
- `ShopScreenTest.kt` — Tests shop interactions and success/failure feedback  
- `ShopViewModelTest.kt` — Unit tests for purchase logic and edge cases  
- `ProfileViewModelTest.kt` — Unit tests for toggles and reward system  

---

## Screenshots

image
 
image

---

## Checklist

- [x] Added full test coverage (UI + ViewModel)  
- [x] Integrated reward system with coins  
- [x] UI matches EduMon theme  
- [x] Tests pass locally (`./gradlew testDebugUnitTest`)  

---

## Notes

- Reward system logic is modular and ready for future Firebase integration.  
- Shop UI supports cosmetic items with prices, ownership state, and feedback animations.  
---

## Issue Reference

Closes #67

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/shop/ShopScreen.kt [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+              if (success) {
+                triggerSuccess() // ✨ animation
+                coroutineScope.launch {
+                  snackbarHostState.showSnackbar("🎉 You purchased ${item.name}!")
+                }
+              } else {
+                triggerFail() // ❌ shake
+                coroutineScope.launch {
+                  snackbarHostState.showSnackbar("❌ Not enough coins to buy ${item.name}")
+                }
```
Comment (Le-Azouz) [inline app/src/main/java/com/android/sample/ui/shop/ShopScreen.kt:60]: A few user-facing strings are hard-coded. Try to keep them in a companion object for refactorability.

File: app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package com.android.sample.ui.shop
+
+import androidx.lifecycle.ViewModel
+import com.android.sample.R
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+// The assistance of an AI tool (ChatGPT) was solicited in writing this test file.
+class ShopViewModel : ViewModel() {
```
Comment (Le-Azouz) [inline app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt:9]: The comment says test file but it is actually production code.

File: app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+   *
+   * @return true if the purchase succeeds, false otherwise.
+   */
+  fun buyItem(item: CosmeticItem): Boolean {
+    val coins = _userCoins.value
+    return if (coins >= item.price && !item.owned) {
+      _userCoins.value = coins - item.price
+      _items.value =
+          _items.value.map { currentItem ->
+            if (currentItem.id == item.id) currentItem.copy(owned = true) else currentItem
```
Comment (Le-Azouz) [inline app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt:28]: Prefer the use of update {} because it is safer and more idiomatic.

File: app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  private val _items = MutableStateFlow(initialCosmetics())
+  val items: StateFlow<List<CosmeticItem>> = _items
+
+  /**
+   * Attempts to buy an item.
+   *
+   * @return true if the purchase succeeds, false otherwise.
+   */
+  fun buyItem(item: CosmeticItem): Boolean {
```
Comment (Le-Azouz) [inline app/src/main/java/com/android/sample/ui/shop/ShopViewModel.kt:22]: If the UI passes a stale CosmeticItem, owned/price could be outdated. It is safer to accept an ID and look up the current item from state.

**Review by Le-Azouz — APPROVED** [review]
Great work implementing the shop screen and updating the Home Screen accordingly. Your work shows great attention to UI detail, theme alignment, and testability. The use of stringResource, shared colors, and modern Compose patterns makes the implementation clean and easy to maintain.
